transformations/speech_disfluency_perturbation/transformation.py

```diff
@@ -0,0 +1,87 @@
+import random
+from typing import List, Optional
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class SpeechDisfluencyPerturbation(SentenceOperation):
+    tasks: List[TaskType] = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+    ]
+    languages: List[str] = ["en"]
+    keywords = [
+        "lexical",
+        "noise",
+        "rule-based",
+        "highly-meaning-preserving",
+        "high-precision",
+        "high-coverage",
+        "high-generations",
+        "social-reasoning",
+    ]
+
+    filler_words: List[str] = ["um", "uh", "erm", "ah", "er"]
+    insert_prob: float
+
+    def __init__(
+        self,
+        seed: int = 0,
+        max_outputs: int = 1,
+        insert_prob: float = 0.2,
+        filler_words: Optional[List[str]] = None,
+    ) -> None:
+        super().__init__(seed, max_outputs=max_outputs)
+        self.insert_prob = insert_prob
+        if filler_words is not None:
+            self.filler_words = filler_words
+
+    def generate(self, sentence: str) -> List[str]:
+        random.seed(self.seed)
+        perturbed_texts = []
+
+        for _ in range(self.max_outputs):
+            tokens = sentence.split()
+            choices = []
+            while not choices:
+                for i in range(1, len(tokens) - 1):
+                    if random.random() < self.insert_prob:
+                        choices.append(i)
```

Comment:
This goes into an infinite loop if `len(tokens) <= 2`

Reply:
Good catch! I've corrected the infinite loop, adjusted the README to note that the results for 1-token inputs might be a bit odd, and added corresponding tests.